lib/rspec/rails/matchers/active_job.rb

```diff
@@ -139,36 +172,78 @@ def queue_adapter
       # @example
       #     expect {
       #       HeavyLiftingJob.perform_later
-      #     }.to have_enqueued_job
+      #     }.to enqueue_job
       #
       #     expect {
       #       HelloJob.perform_later
       #       HeavyLiftingJob.perform_later
-      #     }.to have_enqueued_job(HelloJob).exactly(:once)
+      #     }.to enqueue_job(HelloJob).exactly(:once)
       #
       #     expect {
       #       HelloJob.perform_later
       #       HelloJob.perform_later
       #       HelloJob.perform_later
-      #     }.to have_enqueued_job(HelloJob).at_least(2).times
+      #     }.to enqueue_job(HelloJob).at_least(2).times
       #
       #     expect {
       #       HelloJob.perform_later
-      #     }.to have_enqueued_job(HelloJob).at_most(:twice)
+      #     }.to enqueue_job(HelloJob).at_most(:twice)
       #
       #     expect {
       #       HelloJob.perform_later
       #       HeavyLiftingJob.perform_later
-      #     }.to have_enqueued_job(HelloJob).and have_enqueued_job(HeavyLiftingJob)
+      #     }.to enqueue_job(HelloJob).and enqueue_job(HeavyLiftingJob)
       #
       #     expect {
       #       HelloJob.set(wait_until: Date.tomorrow.noon, queue: "low").perform_later(42)
-      #     }.to have_enqueued_job.with(42).on_queue("low").at(Date.tomorrow.noon)
-      def have_enqueued_job(job = nil)
+      #     }.to enqueue_job.with(42).on_queue("low").at(Date.tomorrow.noon)
+      def enqueue_job(job = nil)
+        unless ::ActiveJob::QueueAdapters::TestAdapter === ::ActiveJob::Base.queue_adapter
+          raise StandardError, "To use enqueue_job matcher set `ActiveJob::Base.queue_adapter = :test`"
```

Comment:
For sure. But I didn't change error class here, it was `StandardError ` before me.